### ISSUE TITLE
New impersonation changes not working in Microsoft Edge for Mac

### ISSUE THREAD

**Issue by foomonkey** [rajyraman/Levelup-for-Dynamics-CRM#266]
Browser Version: [object Navigator]
    Extension Version: 3.6.2
    ----------------------------------------------------------
    Previous impersonation feature worked fine on Microsoft Edge for Mac.  The new implementation (where you have to type the user's primary email address) does not.  When you type in the email address and then toggle the switch to On, it does not refresh the page (like it used to) and does not update the LevelUp extension button to show you are impersonating.

This may not be a Mac specific issue as someone else on my team has reported a similar problem on Windows.

**Comment by rajyraman** [comment]
@foomonkey - Are you using username?

![image](https://user-images.githubusercontent.com/5035266/224459062-ee1ecc15-4f86-4f7f-97b3-0553dcc224a1.png)

 I had to release the extension early because I had to fix the bug that killed the whole extension, so I could not finish my changes around the impersonation feature. But I did check using a guest user and the impersonation worked in my environment.

Can you try this query in FXB and see if it returns any results? If this query returns result, choosing a user and hitting the toggle to reload the tab as the impersonated user.

```xml
<fetch version='1.0' output-format='xml-platform' mapping='logical' distinct='true'>
	<entity name='systemuser'>
		<attribute name='systemuserid'/>
		<attribute name='fullname'/>
		<attribute name='domainname'/>
		<attribute name='azureactivedirectoryobjectid'/>
		<filter>
			<condition attribute='isdisabled' operator='eq' value='0'/>
			<condition attribute='islicensed' operator='eq' value='1'/>
			<condition attribute='accessmode' operator='eq' value='0'/>
			<condition attribute='domainname' operator='eq' value='${impersonateRequest.userName}'/>
		</filter>
		<order attribute='fullname' descending='false'/>
	</entity>
</fetch>
```
I am still making changes to the impersonation feature this week. Below is how it will look in the future.

![image](https://user-images.githubusercontent.com/5035266/224459430-aa60bad6-e56a-496a-9d1f-062ab8c1d4b7.png)

**Comment by rajyraman** [comment]
@foomonkey - Here is a sneak peek of the yet to be released feature.

https://user-images.githubusercontent.com/5035266/224462045-a887677a-a3b1-4aa7-b340-75005f96ff9a.mp4

**Comment by foomonkey** [comment]
@rajyraman, thanks for the insight into the feature.

So to answer your question, yes, I am using the username.  So I've done some more testing and this is what I've come up with:
Edge on Mac: not working (doesn't refresh the page after toggling)
Chrome on Mac: working (refreshes the page as expected)
Edge on Windows: working
Edge on a _different Mac_: working (so it isn't a consistent Mac/Edge issue)
A coworker:
Chrome on Windows: not working (doesn't refresh the page after toggling... so its not just a Mac issue)

We were all using the same username in the same environment across all of these scenarios.
So it seems like there is some situation that is causing the feature to fail on some installs.  I'll try removing the extension from Edge on Mac and re-adding it to see if that fixes it.

**Comment by foomonkey** [comment]
@rajyraman 
Ok, so a couple of things:
1. The version of the extension in the Microsoft Edge store is still the previous one (with the older implementation of Impersonation)
2. I installed the extension from the Chrome Extension store in Edge (which is supported by Edge) and got the new implementation
3. It still does not work on Edge on one of my Macs... so uninstalling and re-installing doesn't fix the issue.

Its unclear if its a state issue (some cached data affecting the feature) or a bug that selectively affects systems.
Any suggestions?  I'm more than happy to try them.
Thanks!

**Comment by rajyraman** [comment]
@foomonkey - I submitted to both Edge and Chrome yesterday. But, no email from Edge yet. They are probably taking more time due to additional permissions.

Did you run the FetchXml and see if it returns any results? If the tab did not refresh, no user was found for that username most likely.

**Comment by rajyraman** [comment]
@foomonkey - You can try clicking the reset button once to flush out the impersonation event handlers.

**Comment by foomonkey** [comment]
@rajyraman 
I just went and tried the fetchxml and it does return the user.  And like I mentioned, it's the same user that works fine with this amazing extension on other browsers/PCs.

So we have a situation where one of my Mac's Edge browser doesn't work **and** a coworkers Windows Chrome browser doesn't work in the same way for the same user (that exists) in the same environment.

Hitting the reset button doesn't appear to have an effect on this issue.

I even considered that maybe the query for the user was just taking a long time (longer than I was giving it) on the problematic system so I entered the username and toggled the impersonation control and just let it sit there to see if, after some time, it would come back and refresh the page... but after a couple of minutes, it never does.

**Comment by rajyraman** [comment]
@foomonkey - That is interesting. If there were some bug it should fail everywhere. Are all the browsers the same in terms of version - version of the browser? You can also check "inspect service worker" on the extension page to see if there are any errors logged in the console.

**Comment by rajyraman** [comment]
@foomonkey - You can try sideloading the latest release (not yet published) and see if this behavior still exists.

https://github.com/rajyraman/Levelup-for-Dynamics-CRM/releases/tag/refs%2Fheads%2Ffeature%2Fimpersonation-search

**Comment by foomonkey** [comment]
@rajyraman Thanks
So I side loaded the newer version and it won't search or let me impersonate a user if I fully type in their username.  The Impersonate button is always grayed out.  None of the other features are working either... god mode, logical names, etc.

I went to the Extensions config and clicked on the Errors link for Level Up and get this:
<img width="841" alt="image" src="https://user-images.githubusercontent.com/100851/224541822-de824c5d-230c-411f-b972-09f52bb72ebc.png">

Here's the actual errors (both are the same):
`Uncaught Error: Module build failed (from ./node_modules/ts-loader/index.js): Error: Cannot find module 'typescript' Require stack: - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/ts-loader/dist/utils.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/ts-loader/dist/after-compile.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/ts-loader/dist/instances.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/ts-loader/dist/index.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/ts-loader/index.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/loader-runner/lib/loadLoader.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/loader-runner/lib/LoaderRunner.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/webpack/lib/NormalModule.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/webpack/lib/NormalModuleFactory.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/webpack/lib/Compiler.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/webpack/lib/webpack.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/tasks/scripts.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/gulpfile.babel.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/gulp/node_modules/gulp-cli/lib/versioned/^4.0.0/index.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/gulp/node_modules/gulp-cli/index.js - /home/runner/work/Levelup-for-Dynamics-CRM/Levelup-for-Dynamics-CRM/node_modules/gulp/bin/gulp.js`

**Comment by rajyraman** [comment]
@foomonkey - The artifact from the GitHub Action seems to be incorrect. Can you try again with https://github.com/rajyraman/Levelup-for-Dynamics-CRM/releases/tag/v3.6.3?

**Comment by rajyraman** [comment]
@foomonkey - Update is now live on Edge.

https://microsoftedge.microsoft.com/addons/detail/level-up-for-dynamics-365/mdjlgdkgmhlmcikdmeehcecolehipicf

**Comment by foomonkey** [comment]
@rajyraman Thanks for all of your help with this.  The update didn't work either... so I finally decided for the nuclear option.
I blew away the cache folder and the application support folder on the Mac for Edge, and then reinstalled Edge. At first, when it synced my user profile, it had the older version of the extension.  But I removed it and re-added it from the Edge store and got the newest published version.

It works now.  So I dont know what happened to my environment in between the previous version and the latest, but something didn't sit well with it.  Oh, one other thing... before I blew away my install, when I updated to the current extension in the edge store, I checked the extension for errors and it said it couldn't register the background process (sorry, dont have the exact error now... it just showed the json or xml for the background process definition).
